### PR TITLE
Introduce OpenSSF Scorecard GitHub action

### PR DESCRIPTION
**Pull request by Stephan202**
Suggested commit message:
```
Introduce OpenSSF Scorecard GitHub action (#574)

And resolve some of the issues it identified.

See https://securityscorecards.dev
```

For the output, see [this view](https://github.com/PicnicSupermarket/error-prone-support/security/code-scanning?query=is%3Aopen+pr%3A574). Only warnings about non-pinned dependencies remain. Those can be resolved using [Renovate](https://docs.renovatebot.com/modules/manager/github-actions/#additional-information), IIUC, so I propose we address that separately.

### CODE REVIEW

File: .github/workflows/pitest-update-pr.yml [hunk 1]
```diff
@@ -9,16 +9,20 @@
...
   actions: read
-  checks: write
-  contents: read
-  pull-requests: write
 jobs:
   update-pr:
     if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    permissions:
+      actions: read
+      checks: write
```
[the remark this answers is not in the document]
Comment (Stephan202) [reply .github/workflows/pitest-update-pr.yml:17]: This code is executed from `master`, thus safe IIUC.

File: .github/workflows/openssf-scorecard.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  analyze:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3.1.0
+        # XXX: Also apply elsewhere?
+        with:
+          persist-credentials: false
+      - name: Run OpenSSF Scorecard analysis
+        uses: ossf/scorecard-action@v2.1.3
```
[the remark this answers is not in the document]
Comment (oxkitsune) [reply .github/workflows/openssf-scorecard.yml:28]:
```suggestion
        uses: ossf/scorecard-action@80e868c13c90f172d68d1f4501dee99e2479f7af # v2.1.3
```
It's complaining about not using a pinned hash, any reason to not do this?
Imo if we keep a "human readable" comment with the actual version we should be good!
[Dependabot even has support](https://github.com/dependabot/dependabot-core/issues/4691) to update this comment!

File: .github/workflows/openssf-scorecard.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3.1.0
+        with:
+          persist-credentials: false
+      - name: Run OpenSSF Scorecard analysis
+        uses: ossf/scorecard-action@v2.1.3
+        with:
+          results_file: results.sarif
+          results_format: sarif
```
Comment (rickie) [inline .github/workflows/openssf-scorecard.yml:31]: Cool stuff! This is what was also used by the EP PR for collecting diagnotics.

File: .github/workflows/openssf-scorecard.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+          publish_results: ${{ github.ref == 'refs/heads/master' }}
+      - name: Update GitHub's code scanning dashboard
+        uses: github/codeql-action/upload-sarif@v2.2.11
+        with:
+          sarif_file: results.sarif
+
```
Comment (rickie) [inline .github/workflows/openssf-scorecard.yml]: One extra line here 👀.

**Review by rickie — APPROVED** [review]
This looks so interesting! Didn't dive too deep into all the checks this has, but it sounds promising.

Nice setup, could spot only 1 tiny thing, so added a commit.

File: .github/workflows/openssf-scorecard.yml (hunk 4 above)
Comment (Stephan202) [reply .github/workflows/openssf-scorecard.yml]: Nice eye for detail! :smile:

**Comment by rickie** [comment]
Oh btw, Monday we'll do some big upgrades to Renovate. Let's see if it picks up on the GHA dependencies again. Otherwise, we'll look into that. Pending that we can setup the pinning that it now flags 😄. So to be continued.

File: .github/workflows/openssf-scorecard.yml (hunk 4 above)
Comment (rickie) [reply .github/workflows/openssf-scorecard.yml]: Consistency for the win 😉.

**Pull request merged**